### PR TITLE
feat:add input component

### PR DESCRIPTION
**Pull request by shubhamkumar648**
add Input component
1. basic input
2. checkbox and radio button
inp1
inp2
inp3

3.with validation and errors style

### CODE REVIEW

File: component/input.html [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+<html lang="en">
+
+<head>
+    <meta charset="UTF-8" />
+    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <link rel="stylesheet" href="../styles/main.css" />
+    <title>GK UI</title>
+    <script src="https://kit.fontawesome.com/a4417b6562.js" crossorigin="anonymous"></script>
+</head>
```
Comment (shuntzhere) [inline component/input.html:11]: Load the script after the body, this won't make any changes here but is considered a good practice.

File: component/input.html [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+            <div class="aside__header">
+                <span class="fs-xl">Component</span>
+            </div>
+
+            <hr />
+
+            <div class="aside__content p-1">
+                <span>
+                    <a class="active link__nostyle" href="../component/avatar.html">Avatar</a></span>
+
```
Comment (shuntzhere) [inline component/input.html]: Indentation is required.

File: component/input.html [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+                <span> <a class="active link__nostyle" href="../component/input.html">Input</a></span>
+
+                <span>
+                    <a class="active link__nostyle" href="../component/textutilities.html">Text Utilities</a></span>
+                <span>
+                    <a class="active link__nostyle" href="../component/navigation.html">Navigation</a></span>
+                <span>
+                    <a class="active link__nostyle" href="../component/snackbar.html">Snackbar</a></span>
+                <span>
+                    <a class="active link__nostyle" href="../component/rating.html">Rating</a></span>
```
Comment (shuntzhere) [inline component/input.html]: I think only the current file needs to have the active class. Check this once.

**Review by shuntzhere — CHANGES_REQUESTED** [review]
Suggested some changes. Make sure to check the active class problem.

File: component/input.html (hunk 3 above)
Comment (shubhamkumar648) [reply component/input.html]: yes i put intensionally for future

File: component/input.html (hunk 2 above)
Comment (shubhamkumar648) [reply component/input.html]: ok i will update in next commit

File: component/input.html (hunk 1 above)
Comment (shubhamkumar648) [reply component/input.html:11]: thanks for suggestion !

**Comment by iamsaurav28** [comment]
hey give some space between naming and inputbox, for eg: email- and its inputbox. so it ll look clean

**Review by shuntzhere — APPROVED** [review]
Looks good then. Just keep in mind that active class is used in every sidebar element.

**Comment by shubhamkumar648** [comment]
> Looks good then. Just keep in mind that active class is used in every sidebar element.

yeah ! i creat one issue branch if somrthing  will happen

**Comment by shubhamkumar648** [comment]
> 

ok thank you suggestion

**Review by iamsaurav28 — APPROVED** [review]

**Comment by shubhamkumar648** [comment]
> hey give some space between naming and inputbox, for eg: email- and its inputbox. so it ll look clean

i  use bem naming convention so i used _

**Pull request merged**